python/lsst/meas/astrom/astrometry.py

```diff
@@ -0,0 +1,257 @@
+from __future__ import absolute_import, division, print_function
+
+import math
+
+import numpy
+
+from lsst.afw.image import DistortedTanWcs, ExposureF
+from lsst.afw.table import Point2DKey
+from lsst.afw.cameraGeom import TAN_PIXELS
+import lsst.pex.config as pexConfig
+import lsst.pipe.base as pipeBase
+from .loadAstrometryNetObjects import LoadAstrometryNetObjectsTask
+from .matchOptimisticB import MatchOptimisticBTask
+from .fitTanSipWcs import FitTanSipWcsTask
+
+class AstrometryConfig(pexConfig.Config):
+    refObjLoader = pexConfig.ConfigurableField(
+        target = LoadAstrometryNetObjectsTask,
+        doc = "reference object loader",
+    )
+    matcher = pexConfig.ConfigurableField(
+        target = MatchOptimisticBTask,
+        doc = "reference object/source matcher",
+    )
+    wcsFitter = pexConfig.ConfigurableField(
+        target = FitTanSipWcsTask,
+        doc = "WCS fitter",
+    )
+    forceKnownWcs = pexConfig.Field(
+        dtype = bool,
+        doc= "Assume that the input image's WCS is correct, without comparing it to any external reality " +
+            "  (but still match reference objects to sources)",
+        default = False,
+    )
+
+class AstrometryTask(pipeBase.Task):
+    """!Match input sourceCat with a reference catalog and solve for the Wcs
+
+    \anchor AstrometryTask_
```

Comment:
You're using `\` for Doxygen tags in this docstring, but the file uses `@` everywhere else.  I have a slight preference for `@` (because it uniquely identifies markup as Doxygen), but I think it's more important to get consistency within a file one way or the other.